sdcm/remote.py

```diff
@@ -963,3 +967,126 @@ def run_output_check(self, cmd, timeout=None, ignore_status=False,
 
         if not ignore_status and result.exit_status > 0:
             raise process.CmdError(command=result.command, result=result)
+
+
+class RemoteFabric(object):
```

Comment:
Nice, even though we had many problems with the original remoter, which was based in Fabric, but turned out to be unreliable.

If now things work fine, that's great.

Reply:
The current remoter based on subprocess wrapping was a solution to the fact that paramiko and fabric were extremely unreliable, that's what I meant.

Reply:
:+1: maybe you remember specific issues that you encountered with Fabric (the issues that made it unreliable) ?

Reply:
Background: Commit https://github.com/scylladb/scylla-cluster-tests/commit/04601b0377e4e5626e294767ff8d4f7574355ddf switched two subprocess.

```
commit 04601b0377e4e5626e294767ff8d4f7574355ddf
Author: Lucas Meneghel Rodrigues <lmr@scylladb.com>
Date:   Tue Feb 2 19:08:51 2016 -0200

    sdcm.remote: Remove paramiko, add new Remote implementation
    
    Since paramiko has key management bugs that were far too
    inconvenient [1], let's ditch it altogether with fabric,
    its wrapper library. Let's introduce a new class based
    around wrapping the actual SSH C clients. It doesn't have
    the conveniences and fine grained error reporting like
    a real python implementation, but it should be good enough
    for our purposes, with no annoying key management bugs.
    
    This commit introduces the library. The API changed slightly,
    so this also changes the callers accordingly.
    
    [1] https://github.com/paramiko/paramiko/issues/387
```

Reply:
I read about this issue before, and as far as I understand this is not relevant for us.
The issue is mainly when using **multiple** keys _encrypted with a passphrase_ and when using SSH agent.
We don't use and will not use such complicated methods of authentication, we just use PKEY to connect.